ui/src/main/java/com/doordeck/sdk/ui/verify/VerifyDeviceActivity.kt

```diff
@@ -193,11 +195,35 @@ internal class VerifyDeviceActivity : BaseActivity(), VerifyDeviceView {
     }
 
     private fun hideVerifySend() {
-        binding.tvReSendCode.setOnClickListener { presenter.onSendCode() }
+        binding.tvReSendCode.setOnClickListener { onSendCode() }
         binding.tvReSendCode.text = getString(R.string.resend_code)
         binding.tvReSendCodeSent.visibility = View.GONE
     }
 
+    private fun onSendCode() {
+        binding.verifyLoading.visibility = View.VISIBLE
+        presenter.onSendCode()
+            .thenApply { userDetails ->
+                Handler(Looper.getMainLooper()).post {
+                    setEmail(userDetails.email)
+                    verifyCodeSuccess()
+                }
+            }
+            .exceptionally { error ->
+                Handler(Looper.getMainLooper()).post {
+                    when (error.cause) {
```

Comment:
Beware: Newer versions do not throw `TooManyRequestsException` on the `assistedRegisterEphemeralKey` call. Instead, the response will contain `requiresRetry = true`

Reply:
I'm chaining CompletableFutures as it's a "custom narrow SDK". What do you think about the change I made?

Reply:
That should work